bin/generate-starter-schema.py

```diff
@@ -92,13 +92,17 @@ def convert_row_to_property(row, sheet_has_autopopulate):# noqa
 
     if field_type.startswith("AMT-"):
         prop["type"] = "number"
+        prop["minimum"] = 0
+        prop["maximum"] = int('9' * int(field_type.split('-')[1]))
 
     if field_type.startswith("NUM-"):
-        prop["type"] = "integer"
+        length = field_type.split('-')[1].strip()
+        prop["type"] = "string"
+        prop["minLength"] = 0
+        prop["maxLength"] = int(length)
+        prop["pattern"] = f'^\d{{0,{length}}}$'
```

Comment:
I believe we want to enforce the full length of the string so the regex should be '^\d{{length}}$'